src/state/selectors.ts

```diff
@@ -7,47 +7,36 @@ import type { ReduxState } from './reducer'
 import type { LpId, StakingId, UserStakingId } from './slices/opportunitiesSlice/types'
 import type { PubKey } from './slices/validatorDataSlice/validatorDataSlice'
 
-// List of all the params filter consumed with selectParamFromFilter
-type ParamFilter = {
+/**
+ * List of all the params filter consumed with selectParamFromFilter
+ *
+ * note - this **must** stay as Partial, it's the selectors consumers responsibility to check
+ * for existence of these. it's neither ergonomic nor feasible for the view layer consumers
+ * to guard, or even necessarily have, the required params before calling selectors.
+ *
+ * wen conditional hooks?
+ */
+type ParamFilter = Partial<{
   accountAddress: string
   contractAddress: string
   assetId: AssetId
   accountId: AccountId
+  validatorAddress: PubKey
   accountNumber: number
   chainId: ChainId
-  validatorAddress: PubKey
   userStakingId: UserStakingId
   stakingId: StakingId
   lpId: LpId
-}
-
-// List of all the params filter consumed with selectParamFromFilterOptional
-type OptionalParamFilter = {
-  accountAddress?: string
-  contractAddress?: string
-  assetId?: AssetId
-  accountId?: AccountId
-  validatorAddress?: PubKey
-}
+}>
 
 type ParamFilterKey = keyof ParamFilter
-type OptionalParamFilterKey = keyof OptionalParamFilter
 
 export const selectParamFromFilter = <T extends ParamFilterKey>(param: T) =>
   createCachedSelector(
-    (_state: ReduxState, filter: Pick<ParamFilter, T>): ParamFilter[T] | '' =>
-      filter?.[param] ?? '',
+    (_state: ReduxState, filter: Pick<ParamFilter, T>): ParamFilter[T] | undefined =>
```

Comment:
all filters now return e.g. `AccountId | undefined` - forcing consuming selectors to check for them, and allowing view more flexibility in calling them. it's the caller's responsibility to ensure they're calling selectors with the correct parameters.